src/components/Dropdown/index.js

```diff
@@ -3,15 +3,43 @@ import * as React from 'react'
 
 type Props = {
   onToggle: () => any,
-  children: ({ isOpened: boolean, toggle: () => any }) => React.Node
+  onCancel: () => any,
+  onApply: () => any,
+  children: ({
+    isOpened: boolean,
+    toggle: () => any,
+    cancel?: () => any,
+    apply?: () => any
+  }) => React.Node
 }
 
 type State = {
   isOpened: boolean
 }
 
 class Dropdown extends React.Component<Props, State> {
+  static defaultProps = {
+    onToggle: () => {},
+    onCancel: () => {},
+    onApply: () => {}
+  }
+
   state = { isOpened: false }
+
+  cancel = () => {
+    this.setState(
+      ({ isOpened }) => ({ isOpened: !isOpened }),
+      () => this.props.onCancel()
```

Comment:
can’t you use `this.setState({ isOpened } => ({ isOpened: ! isOpened }), this.props.onCancel)`?
Or does that cause context issues?